habitat/plan.sh

```diff
@@ -70,6 +69,8 @@ do_prepare() {
   export OPENSSL_INCLUDE_DIR="$(pkg_path_for openssl)/include"
   export SSL_CERT_FILE="$(pkg_path_for cacerts)/ssl/cert.pem"
   export CPPFLAGS="${CPPFLAGS} ${CFLAGS}"
+  set_runtime_env LD_LIBRARY_PATH "$LD_RUN_PATH"
```

Comment:
This and line 101 seem fairly generic changes that don't explain how this is implementing the stated goal of the PR headline.

Reply:
I know there is more detailed info in the JIRA ticket which I cannot seem to find. @smandaka could you provide the link?

Reply:
https://chefio.atlassian.net/browse/CHEF-660

Reply:
See https://chefio.atlassian.net/browse/CHEF-660?focusedCommentId=82561 for an explanation of `LD_LIBRARY_PATH`. The linked Jira comment had a different solution using `bundle config --local build.ffi "--with-ldflags=-Wl,-rpath=${LD_RUN_PATH}"` which only addresses the native ffi gem. The `LD_LIBRARY_PATH` accomplishes the same goal but could apply to any native gem.

Reply:
Sounds good Matt. @smandaka  pls baseline the plan to match what matt had originally done. Subsequent to that we can incrementally bring in the changes one by one recommended by Matt and Johny and iterate on build cycle incrementally.